### PR TITLE
Update the configuration files for testing against non-openj9 SDK

### PR DESCRIPTION
**Pull request by dutingda**
- Updating build.xml to make compilation part work for non-openj9 SDK
- Updating playlist.xml to restrict several tests to only openj9 SDK

[ci skip]

Signed-off-by: Tingda Du <dutingda@outlook.com>

### CODE REVIEW

**Comment by dutingda** [comment]
@llxia @smlambert Please review thanks!

**Comment by smlambert** [comment]
Thanks @dutingda - please see https://github.com/eclipse/openj9/blob/master/CONTRIBUTING.md#legal-considerations for instructions on signing the [ECA (Eclipse Contributors Agreement)](http://www.eclipse.org/legal/ECA.php) which will need to be done before your contribution would be merged.

File: test/functional/cmdLineTests/build.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 
 <!--
-  Copyright (c) 2016, 2018 IBM Corp. and others
+  Copyright (c) 2016, 2018, 2019 IBM Corp. and others
```
Comment (smlambert) [inline test/functional/cmdLineTests/build.xml]: change to 
Copyright (c) 2016, 2019 IBM Corp. and others

Comment (dutingda) [reply test/functional/cmdLineTests/build.xml]: Have fixed thanks

File: test/functional/DDR_Test/build.xml [hunk 2]
```diff
@@ -99,13 +100,25 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+						<antcall target="clean" inheritall="true" />
+					</then>
+				</if>
 			</then>
 		</if>
 	</target>
 </project>
+
+
+
```
Comment (renfeiw) [inline test/functional/DDR_Test/build.xml]: Please remove the extra spaces.

File: test/functional/JLM_Tests/build.xml [hunk 3]
```diff
@@ -114,7 +115,15 @@
...
+	<target name="build" >
+		<if>
+      <or>
+        <equals arg1="${JDK_IMPL}" arg2="ibm"  />
+        <equals arg1="${JDK_IMPL}" arg2="openj9" />
+      </or>
+      <then>
+        <antcall target="clean" inheritall="true" />
+      </then>
+    </if>
```
Comment (renfeiw) [inline test/functional/JLM_Tests/build.xml]: Please fix the indentation.

File: test/functional/Java8andUp/build.xml [hunk 4]
```diff
@@ -339,7 +339,15 @@
...
+	<target name="build" >
+		<if>
+      <or>
+        <equals arg1="${JDK_IMPL}" arg2="ibm"  />
+        <equals arg1="${JDK_IMPL}" arg2="openj9" />
+      </or>
+      <then>
+        <antcall target="clean" inheritall="true" />
+      </then>
+    </if>
```
Comment (renfeiw) [inline test/functional/Java8andUp/build.xml]: Please fix the indentation.

File: test/functional/JavaAgentTest/build.xml [hunk 5]
```diff
@@ -104,7 +104,15 @@
...
-		<antcall target="clean" inheritall="true" />
+		<if>
+            <or>
+                <equals arg1="${JDK_IMPL}" arg2="ibm"  />
+                <equals arg1="${JDK_IMPL}" arg2="openj9" />
+            </or>
+            <then>
+            	<antcall target="clean" inheritall="true" />
+             </then>
+       </if>
```
Comment (renfeiw) [inline test/functional/JavaAgentTest/build.xml]: Please fix the indentation.

File: test/functional/UnsafeTest/build.xml [hunk 6]
```diff
@@ -94,15 +94,23 @@
...
+							<equals arg1="${JDK_VERSION}" arg2="9" />
+							<equals arg1="${JCL_VERSION}" arg2="current" />
+						</and>
+					</not>
+					<then>
+						<antcall target="clean" inheritall="true" />
+					</then>
+				</if>
+             </then>
+       </if>
```
Comment (renfeiw) [inline test/functional/UnsafeTest/build.xml]: Please fix the indentation.

File: test/functional/VM_Test/build.xml [hunk 7]
```diff
@@ -173,6 +173,14 @@
...
-		<antcall target="clean" inheritall="true" />
+		<if>
+            <or>
+                <equals arg1="${JDK_IMPL}" arg2="ibm"  />
+                <equals arg1="${JDK_IMPL}" arg2="openj9" />
+            </or>
+            <then>
+            	<antcall target="clean" inheritall="true" />
+             </then>
+       </if>
```
Comment (renfeiw) [inline test/functional/VM_Test/build.xml]: Please fix the indentation.

File: test/functional/cmdLineTests/build.xml [hunk 8]
```diff
@@ -38,20 +38,28 @@
...
+						</subant>
+					</then>
+					<else>
+						<subant target="">
+							<fileset dir="." includes="*/build.xml" />
+						</subant>
+					</else>
+				</if>			
+             </then>
+       </if>
```
Comment (renfeiw) [inline test/functional/cmdLineTests/build.xml]: Please fix the indentation.

**Comment by smlambert** [comment]
Please let me know when ready for re-review, you also will need to fix the merge conflict.

File: test/functional/Jsr292/playlist.xml [hunk 9]
```diff
@@ -206,12 +221,41 @@
...
+	$(TEST_STATUS)</command>
+		<levels>
+			<level>extended</level>
+		</levels>
+		<groups>
+			<group>functional</group>
+		</groups>
+		<subsets>
+			<subset>9+</subset>
+		</subsets>
```
Comment (llxia) [inline test/functional/Jsr292/playlist.xml]: Please add:
```
<impls>
	<impl>hotspot</impl>
</impls>
```

File: test/functional/Jsr292/playlist.xml [hunk 10]
```diff
@@ -100,6 +100,13 @@
 		<subsets>
 			<subset>11+</subset>
 		</subsets>
+		<!-- 
+			TODO: Further investigation is needed running against non-openj9 sdk
```
Comment (llxia) [inline test/functional/Jsr292/playlist.xml]: Please update the message to:
TODO: Further investigation is needed `for` running against non-openj9 sdk

File: test/functional/cmdLineTests/build.xml [hunk 11]
```diff
@@ -51,7 +51,7 @@
 					<fileset dir="." includes="*/build.xml" />
 				</subant>
 			</else>
-		</if>
+		</if>
```
Comment (llxia) [inline test/functional/cmdLineTests/build.xml]: There is no change for this file other than the extra space and the copyright year. Please revert.

File: test/functional/cmdline_options_testresources/src/Dump.java [hunk 12]
```diff
@@ -1,67 +0,0 @@
-/*******************************************************************************
```
Comment (llxia) [inline test/functional/cmdline_options_testresources/src/Dump.java]: Why this file is removed?

File: test/functional/cmdline_options_testresources/src_90/com/ibm/j9/getcallerclass/GetCallerClassTests.java [hunk 13]
```diff
@@ -1,254 +0,0 @@
-/*******************************************************************************
```
Comment (llxia) [inline test/functional/cmdline_options_testresources/src_90/com/ibm/j9/getcallerclass/GetCallerClassTests.java]: Why this file is removed?

**Comment by llxia** [comment]
Please fix all indentation issue in the PR.

File: test/functional/cmdLineTests/jvmtitests/build.xml [hunk 14]
```diff
@@ -126,7 +126,15 @@
...
+	<target name="build" >
+		<if>
+      		<or>
+        		<equals arg1="${JDK_IMPL}" arg2="ibm"  />
+        		<equals arg1="${JDK_IMPL}" arg2="openj9" />
+      		</or>
+      		<then>
+        		<antcall target="clean" inheritall="true" />
+      		</then>
+    	</if>
```
Comment (llxia) [inline test/functional/cmdLineTests/jvmtitests/build.xml]: Please fix the indentation.

File: test/functional/cmdLineTests/pageAlignDirectMemory/playlist.xml [hunk 15]
```diff
@@ -37,5 +37,9 @@
 		<groups>
 			<group>functional</group>
 		</groups>
+    <impls>
+			<impl>openj9</impl>
+			<impl>ibm</impl>
+		</impls>
```
Comment (llxia) [inline test/functional/cmdLineTests/pageAlignDirectMemory/playlist.xml]: Please fix the indentation.

File: test/functional/cmdLineTests/shareClassTests/DataHelperTests/build.xml [hunk 16]
```diff
@@ -126,6 +126,14 @@
...
-		<antcall target="clean" inheritall="true" />
+		<if>
+      		<or>
+        		<equals arg1="${JDK_IMPL}" arg2="ibm"  />
+        		<equals arg1="${JDK_IMPL}" arg2="openj9" />
+      		</or>
+      		<then>
+        		<antcall target="clean" inheritall="true" />
+      		</then>
+    	</if>
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/DataHelperTests/build.xml]: Please fix the indentation.

File: test/functional/cmdLineTests/shareClassTests/SCHelperCompatibilityTests/build.xml [hunk 17]
```diff
@@ -146,6 +146,14 @@
...
-		<antcall target="clean" inheritall="true" />
+		<if>
+      		<or>
+        		<equals arg1="${JDK_IMPL}" arg2="ibm"  />
+        		<equals arg1="${JDK_IMPL}" arg2="openj9" />
+      		</or>
+      		<then>
+        		<antcall target="clean" inheritall="true" />
+      		</then>
+    	</if>
```
Comment (llxia) [inline test/functional/cmdLineTests/shareClassTests/SCHelperCompatibilityTests/build.xml]: Please fix the indentation.

**Comment by llxia** [comment]
Still have error `Unrecognized option...`

https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/411/tapResults/

**Comment by llxia** [comment]
Please rebase and fix the merge conflict. Thanks.

**Comment by llxia** [comment]
Please update the title and description to be more meaningful and fix grammar errors.
For example, the title could be `Update the configuration files for testing against non-openj9 SDK`

**Comment by llxia** [comment]
Please also update the description/commit msg to fix the grammar errors

**Comment by llxia** [comment]
Please fix the `Unrecognized option` error:
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/425/tapResults/

**Comment by llxia** [comment]
Failure in `sanity.functional` in JDK11 hotspot.
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/428/tapResults/

File: test/functional/DDR_Test/build.xml [hunk 18]
```diff
@@ -99,12 +100,21 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 	<!-- Temporarily disable compilation on zos until ddr is supported; github.com/eclipse/openj9/issues/1511 -->
 	<target name="build" >
 		<echo>os.name: ${os.name}</echo>
+		<echo>impl: ${JDK_IMPL}</echo>
```
Comment (llxia) [inline test/functional/DDR_Test/build.xml]: Please remove the echo msg.

**Comment by llxia** [comment]
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/445/
https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/444/

**Comment by llxia** [comment]
jenkins test sanity,extended all jdk11

**Comment by llxia** [comment]
Please fix the errors in PR builds.
```
00:28:18.194  /home/u0020236/workspace1/Test_openjdk11_j9_extended.functional_ppc64_aix_Personal/openjdk-tests/functional/cmdLineTests/stackSizeInfoTest/build.xml:49: The following error occurred while executing this line:
00:28:18.194  Target "clean" does not exist in the project "cmdLineTests".
```

**Comment by llxia** [comment]
jenkins test sanity,extended all jdk11

**Comment by llxia** [comment]
jenkins test sanity,extended all jdk8

**Comment by llxia** [comment]
AIX builds failure are related to https://github.com/eclipse/openj9/issues/4643 and https://github.com/eclipse/openj9/issues/3081

**Pull request merged**